controllers/topolvm_snapshotClass.go

```diff
@@ -0,0 +1,103 @@
+	return vscName
+}
+
+//+kubebuilder:rbac:groups=snapshot.storage.k8s.io,resources=volumesnapshotclasses,verbs=get;create;delete;watch;list
+
+func (s topolvmVolumeSnapshotClass) ensureCreated(r *LVMClusterReconciler, ctx context.Context, lvmCluster *lvmv1alpha1.LVMCluster) error {
+
+	// one volume snapshot class for every deviceClass based on CR is created
+	topolvmSnapshotClasses := getTopolvmSnapshotClasses(lvmCluster)
+	for _, vsc := range topolvmSnapshotClasses {
+
+		// we anticipate no edits to volume snapshot class
+		result, err := cutil.CreateOrUpdate(ctx, r.Client, vsc, func() error { return nil })
+		if err != nil {
+			r.Log.Error(err, "topolvm volume snapshot class reconcile failure", "name", vsc.Name)
+			return err
+		} else {
+			r.Log.Info("topolvm volume snapshot class", "operation", result, "name", vsc.Name)
+		}
+	}
+	return nil
+}
+
+func (s topolvmVolumeSnapshotClass) ensureDeleted(r *LVMClusterReconciler, ctx context.Context, lvmCluster *lvmv1alpha1.LVMCluster) error {
+
+	// construct name of volume snapshot class based on CR spec deviceClass field and
+	// delete the corresponding volume snapshot class
+	for _, deviceClass := range lvmCluster.Spec.Storage.DeviceClasses {
+		vsc := &snapapi.VolumeSnapshotClass{}
+		vscName := fmt.Sprintf("odf-lvm-%s", deviceClass.Name)
+		err := r.Client.Get(ctx, types.NamespacedName{Name: vscName}, vsc)
+
+		if err != nil {
+			// already deleted in previous reconcile
+			if errors.IsNotFound(err) {
+				r.Log.Info("topolvm volume snapshot class is deleted", "VolumeSnapshotClass", vscName)
+				return nil
+			}
+			r.Log.Error(err, "failed to retrieve topolvm volume snapshot class", "VolumeSnapshotClass", vscName)
+			return err
+		}
+
+		// VolumeSnapshotClass exists, initiate deletion
+		if vsc.GetDeletionTimestamp().IsZero() {
+			if err = r.Client.Delete(ctx, vsc); err != nil {
+				r.Log.Error(err, "failed to delete topolvm volume snapshot class", "VolumeSnapshotClass", vscName)
+				return err
+			} else {
+				r.Log.Info("initiated topolvm volume snapshot class deletion", "VolumeSnapshotClass", vscName)
+			}
+		} else {
+			// return error for next reconcile to confirm deletion
+			return fmt.Errorf("topolvm volume snapshot class %s is already marked for deletion", vscName)
+		}
+	}
+	return nil
+}
+
+func (s topolvmVolumeSnapshotClass) updateStatus(r *LVMClusterReconciler, ctx context.Context, lvmCluster *lvmv1alpha1.LVMCluster) error {
+	// intentionally empty as there'll be no status field on VolumeSnapshotClass resource
+	return nil
+}
+
+func getTopolvmSnapshotClasses(lvmCluster *lvmv1alpha1.LVMCluster) []*snapapi.VolumeSnapshotClass {
+	vsc := []*snapapi.VolumeSnapshotClass{}
+
+	for _, deviceClass := range lvmCluster.Spec.Storage.DeviceClasses {
+		snapshotClass := &snapapi.VolumeSnapshotClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("odf-lvm-%s", deviceClass.Name),
+			},
+
+			Driver:         "topolvm.cybozu.com",
```

Comment:
- `Driver: TopolvmCSIDriverName` will be better and works without any issues
- rationale being storageclass, snapclass refers to same csidriver for performing their objective

Reply:
done.